### PR TITLE
Issue #441 Lustre AWS FSx [2]: mount to run

### PR DESCRIPTION
**Pull request by Wedds**
This PR is related to issue #441.
It brings the second part of the Lustre FS support:

- add Lustre modules installation to `install_nfs_client`
- modify `mount_storage.py` to support Lustre fstype

### CODE REVIEW

File: workflows/pipe-common/shell/install_nfs_client [hunk 1]
```diff
@@ -23,11 +23,28 @@ IS_RPM_BASED=$?
...
-    INSTALL_CMD='apt-get install -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" nfs-common cifs-utils -y -qq'
+    CHECK_CMD="dpkg -l | grep nfs-common && dpkg -l | grep cifs-utils && dpkg -l | grep lustre-client"
+    AMAZON_FSX_PUBLIC_KEY_URL="https://fsx-lustre-client-repo-public-keys.s3.amazonaws.com/fsx-ubuntu-public-key.asc"
+    LUSTRE_CLIENT_REPO_URL="https://fsx-lustre-client-repo.s3.amazonaws.com/$(. /etc/os-release;echo $ID $VERSION_CODENAME)"
+    read -r -d '' INSTALL_CMD <<- EOM
+      apt-get install wget apt-utils -y -qq  &&
+      wget -qO- $AMAZON_FSX_PUBLIC_KEY_URL | apt-key add - &&
+      echo "deb $LUSTRE_CLIENT_REPO_URL main" > /etc/apt/sources.list.d/fsxlustreclientrepo.list && apt-get update -qq &&
+      mkdir -p /lib/modules/$(uname -r) &&
+      apt-get install -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" nfs-common cifs-utils linux-aws lustre-client-modules-aws -y > /dev/null
```
Comment (tcibinan) [inline workflows/pipe-common/shell/install_nfs_client]: Could you please double check that these packages `linux-aws lustre-client-modules-aws` can be installed on non-aws deployment runs without errors.

File: workflows/pipe-common/shell/install_nfs_client [hunk 2]
```diff
@@ -23,31 +23,86 @@ IS_RPM_BASED=$?
...
+    else
+        pipe_log_info "--> NFS client [nfs, smb] installed already" "$NFS_INSTALL_TASK"
     fi
-    pipe_log_info "--> NFS client installed" "$NFS_INSTALL_TASK"
-else
-    pipe_log_info "--> NFS client is already installed" "$NFS_INSTALL_TASK"
+    if [ $IS_LUSTRE_INSTALLED -ne 0 ]
+    then
+        pipe_log_info "--> Installing [lustre] client" "$NFS_INSTALL_TASK"
+        LUSTRE_INSTALLATION_LOG="/run/lustre_client_installation.log"
```
Comment (tcibinan) [inline workflows/pipe-common/shell/install_nfs_client]: Let's write logs to a little bit more appropriate location like `/var/log/lustre_client_installation.log`.

File: workflows/pipe-common/shell/install_nfs_client [hunk 3]
```diff
@@ -23,31 +23,86 @@ IS_RPM_BASED=$?
...
+        else
+            pipe_log_info "--> NFS client packages [lustre] installed successfully" "$NFS_INSTALL_TASK"
+        fi
+    else
+        pipe_log_info "--> NFS client packages [lustre] installed already" "$NFS_INSTALL_TASK"
+    fi
+    eval "df -t lustre" > /dev/null 2>&1
+    if [[ -z $(cat /proc/filesystems | grep lustre) ]]; then
+        pipe_log_warn "--> [lustre] filesystem is not available. " "$NFS_INSTALL_TASK"
+    fi
```
Comment (tcibinan) [inline workflows/pipe-common/shell/install_nfs_client]: I suppose there is no need to explicitly check for lustre availability. If lustre doesn't work in container then the corresponding shares mounting will just fail anyway with more precise logs.

File: workflows/pipe-common/shell/install_nfs_client (hunk 2 above)
Comment (Wedds) [reply workflows/pipe-common/shell/install_nfs_client]: Done

File: workflows/pipe-common/shell/install_nfs_client (hunk 3 above)
Comment (Wedds) [reply workflows/pipe-common/shell/install_nfs_client]: Done

**Review by tcibinan — APPROVED** [review]

**Pull request merged**